crypto/init.c

```diff
@@ -729,3 +683,52 @@ int OPENSSL_atexit(void (*handler)(void))
     return 1;
 }
 
+/*
+ * Library  destructor (a.k.a. _fini()). Code here is derived
+ * from stuff found in providers/fips/self_test.c
+ *
+ * We do kind of  best effort cleanup. If given platform/compiler
+ * does not support module destructors then its up to application
+ * to explicitly call OPENSSL_cleanup();
+ */
+#if defined(__GNUC__) && !defined(_AIX)
+
+void ossl_call_cleanup(void);
+
+void __attribute__((destructor))
+ossl_call_cleanup(void)
+{
+    if (ossl_no_cleanup == 0)
+        OPENSSL_cleanup();
+}
+
+#elif (defined(__sun) || defined(_AIX)) && !defined(__GNUC__)
+void _cleanup(void);
+# pragma fini(_cleanup)
+void _cleanup(void)
```

Comment:
This should be named ossl_call_cleanup() unless the symbol must be named _cleanup.

And similarly for the __hpux case below

Reply:
OK will rename it. I was not sure of it I just took it from fips module as-is.